server/hotrod/src/test/scala/org/infinispan/server/hotrod/test/HotRodTestingUtil.scala

```diff
@@ -91,6 +94,19 @@ object HotRodTestingUtil extends Log {
             cfg.transaction().syncCommitPhase(false).syncRollbackPhase(false)
             cfg
          }
+
+         override def getInitializer: ChannelInitializer[Channel] = {
+            // Pass by name since we have circular dependency
+            def getTransport() = {
+               transport
+            }
+            if (configuration.idleTimeout > 0)
+               new HotRodChannelInitializer(this, getTransport(), getEncoder, "test")
+                 with TimeoutEnabledChannelInitializer with SingleByteFrameDecoderChannelInitializer
```

Comment:
Need to add a system property to only enable SingleByteFrameDecoderChannelInitializer for tests we want to verify decoder is working.  For perf tests this will be a big burden.